app/lib/katello/capsule_content.rb

```diff
@@ -13,7 +13,26 @@ def lifecycle_environments(organization_id = nil)
       scope
     end
 
-    def pulp_repos(environments = lifecycle_environments, content_view = nil)
+    def pulp_server
+      uri = URI.parse(self.capsule.url)
+
+      Runcible::Instance.new(
+        :url      => "#{uri.scheme}://#{uri.host.downcase}",
+        :api_path => "/pulp/api/v2/",
+        :user     => "admin",
+        :timeout      => SETTINGS[:katello][:rest_client_timeout],
+        :open_timeout => SETTINGS[:katello][:rest_client_timeout],
+        :cert_auth => {:ssl_client_cert => ::Katello::Cert::Ca.ssl_client_cert,
+                       :ssl_client_key => ::Katello::Cert::Ca.ssl_client_key},
+        :logging  => {:logger     => ::Foreman::Logging.logger('katello/pulp_rest'),
+                      :exception  => true,
+                      :info       => true,
+                      :debug      => true }
+      )
+    end
```

Comment:
Is there any reason contacting Pulp on the main server couldn't follow this same flow of using certificates for authentication? I ask in part because of how this meshes well with some changes I am proposing (https://github.com/Katello/katello/pull/5741/files) around centralizing our generation of the Runcible instance because we have no need of encapsulating that in a thread session. I don't want to propose radical changes to your PR, but if we can coverge toward a similar solution that would be good I feel.